### DISCUSSION TITLE
segysak v0.5 seems to break bokeh output in JupyterLab

### DISCUSSION THREAD

**Discussion by espenraknes** [trhallam/segysak#129] [Help] [answered]
We've been using segysak as a backend for several SGY loaders that we have for interactive work with Jupyter lab. We use this in combination with some dashboards that we have developed using Holoviews/Bokeh. When I tried to update segysak to 0.5.0, it breaks the output from any bokeh stuff that we are using. If I change backend on the dashboards to, for instance matplotlib, it output the dashboard but without the interactivity that bokeh is giving, so that is not an option. Anyone having an idea what is causing this troubles?

**Comment by trhallam** [comment]
Hi @espenraknes , can you please provide some examples of how you are using `segysak` that are broken? Did you simply upgrade the library or did you change to the new loading implementation?

If you are using the new loader, certain changes may affect your dashboards:
 - The data is loaded lazily, and downstream products may require you to actively load it into memory.
 - Coordinate scaling is not applied by default.

If you are using the old loader, the primary breaking change was to the `silient` keyword argument which is now handled  by `Progress` class.

There is a guide to the key changes in the updated [documentation](https://trhallam.github.io/segysak/latest/upgrading.html).

**Reply by espenraknes** [reply]
It actually happens using both loaders. I have made a simple example script here:

```
import xarray as xr
import segysak

from segysak.progress import Progress
Progress.set_defaults(disable=True)

import holoviews as hv
from holoviews import opts

hv.extension('bokeh')

sgy = xr.open_dataset(
    'test.sgy',
        dim_byte_fields={"tracenum":13},
            extra_byte_fields={"cdp":21}
            )

img = hv.Image(sgy.data.values)
img
```
In my jupyter lab (v4.0.8, holoviews=1.18.1) this does not show up. If I change the hv.extension to 'matplotlib' a figure will appear. Segysak v0.4 does not have this issue. I wonder if this is due to the the new loader tool that is implemented in v0.5 and that conflicts with how bokeh is providing to the notebook?

**Reply by trhallam** [reply]
I've not been able to replicate your problem I'm afraid, using my example data the holoview appears renders ok.

I can only suggest you check your environment. I have attached the versions of my working environment.

[env.txt](https://github.com/user-attachments/files/16085352/env.txt)

![Screenshot 2024-07-03 140551](https://github.com/trhallam/segysak/assets/8454087/5fae48de-9bc7-4858-a852-14d5acc36429)

**Reply by espenraknes** [reply, marked as answer]
Ok, thanks for looking into this; i'll check my environments.
